haskell/nettest/Permit.hs

```diff
@@ -0,0 +1,160 @@
+
+  comment "Issue a permit for Revoke"
+  issuePermit pauserAlias Pause
+  let revokeParam = [(mkPermitHash Pause, pauser)]
+  issuePermit pauserAlias (Revoke revokeParam)
+  callFrom (addressResolved user1) contract (Call @"Revoke") revokeParam
+  callFrom (addressResolved user1) contract (Call @"Pause") ()
+    `expectFailure` NettestFailedWith contract [mt|NOT_PAUSER|]
+
+  comment "Issue a permit for Transfer"
+  let transferParam = [ (#from_ .! owner1, #txs .! [(#to_ .! user1, (#token_id .! 0, #amount .! 1))]) ]
+  issuePermit owner1Alias (Call_FA2 $ FA2.Transfer transferParam)
+  callFrom (addressResolved user1) contract (Call @"Transfer") transferParam
+
+  comment "Issue a permit for Update_operators"
+  let param = [ FA2.Add_operator (#owner .! owner1, #operator user1 ) ]
+  issuePermit owner1Alias (Call_FA2 $ FA2.Update_operators param)
+  callFrom (addressResolved user1) contract (Call @"Update_operators") param
+
+  comment "Issue a permit for Configure_minter"
+  let configureMinterParam =
+        ( #minter .! user1
+        , ( #current_minting_allowance .! Nothing
+          , #new_minting_allowance .! 30
+          ))
+  issuePermit masterMinterAlias (Configure_minter configureMinterParam)
+  callFrom (addressResolved user1) contract (Call @"Configure_minter") configureMinterParam
+
+  comment "Issue a permit for Remove_minter"
+  let removeMinterParam = user1
+  issuePermit masterMinterAlias (Remove_minter removeMinterParam)
+  callFrom (addressResolved user1) contract (Call @"Remove_minter") removeMinterParam
+
+  comment "Issue a permit for Set_transferlist"
+  issuePermit owner1Alias (Set_transferlist Nothing)
+  callFrom (addressResolved user1) contract (Call @"Set_transferlist") Nothing
+
+  comment "Issue a permit for Change_pauser"
+  issuePermit owner1Alias (Change_pauser owner2)
+  callFrom (addressResolved user1) contract (Call @"Change_pauser") owner2
+
+  comment "Issue a permit for Change_master_minter"
+  issuePermit owner1Alias (Change_master_minter owner2)
+  callFrom (addressResolved user1) contract (Call @"Change_master_minter") owner2
+
+  comment "Issue a permit for Transfer_ownership"
+  issuePermit owner1Alias (Transfer_ownership owner2)
+  callFrom (addressResolved user1) contract (Call @"Transfer_ownership") owner2
+
+  comment "Issue a permit for Accept_ownership"
+  issuePermit owner2Alias Accept_ownership
+  callFrom (addressResolved user1) contract (Call @"Accept_ownership") ()
+
+  where
+    createUser :: MonadNettest caps base m => Alias -> m (Alias, Address)
+    createUser alias = do
+      addr <- newAddress alias
+      -- Note: `newAddress` prepends the alias with the prefix from the
+      -- `MorleyClientConfig`.
+      -- So we use `getAlias` to get the actual alias (prefix included)
+      -- associated with this address.
+      actualAlias <- getAlias (AddressResolved addr)
+      pure (actualAlias, addr)
+
+    getChainId :: MonadNettest caps base m => m ChainId
+    getChainId = do
```

Comment:
Wonder if we should add a method to fetch chain id to Nettest interface. AFAIU there is a Tezos RPC available to fetch it, so it should be possible. International interface already something to set it, (but it appears to be deprecated now).

Reply:
Yeah, I created a ticket for cleveland here: https://gitlab.com/morley-framework/morley/-/issues/356
And a follow-up ticket for us to update these tests once the cleveland ticket is done: https://github.com/tqtezos/stablecoin/issues/109